lex/lexer.mll

```diff
@@ -214,9 +225,20 @@ and string = parse
           "illegal backslash escape in string: '\\%c%c%c'" c d u) ;
       store_string_char (Char.chr v);
       string lexbuf }
- | '\\' 'x' (['0'-'9' 'a'-'f' 'A'-'F'] as d) (['0'-'9' 'a'-'f' 'A'-'F'] as u)
+  | '\\' 'o' (['0'-'3'] as c) (['0'-'7'] as d) (['0'-'7'] as u)
+    { store_string_char (char_for_octal_code c d u);
+      string lexbuf }
+  | '\\' 'x' (['0'-'9' 'a'-'f' 'A'-'F'] as d) (['0'-'9' 'a'-'f' 'A'-'F'] as u)
     { store_string_char (char_for_hexadecimal_code d u) ;
       string lexbuf }
+  | '\\' 'u' '{' (['0'-'9' 'a'-'f' 'A'-'F'] + as s) '}'
+    { let v = hexadecimal_code s in
+      if in_pattern () && not (Uchar.is_valid v) then
+       warning lexbuf
+        (Printf.sprintf
+          "illegal uchar escape in string: '\\u{%s}'" s);
+      store_string_uchar (Uchar.unsafe_of_int v);
```

Comment:
Unless I'm mistaken, this will `assert false` in buffer.ml (after printing the warning) if `v` is too large or negative.

Reply:
I'll make the warning an error. Decimal escape sequences have the same problem (e.g. "\256").